### PR TITLE
Reworked config

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
- Move config meta data into REST
- seperate views for config groups

TODO:
- [x] Better UI @tsiegleauq
- [x] Back button
- [x] Reset to default (Server request)
- [x] Bulk save for a subpage (Server request; But the UI should be
there first)

Sean: Please also add a dot-menu with the entry "reset to default" for
the complete list and every subpage

### CODE REVIEW

File: client/src/app/core/repositories/config/config-repository.service.ts [hunk 1]
```diff
@@ -13,27 +13,7 @@ import { Identifiable } from 'app/shared/models/base/identifiable';
...
-     */
-    config: ViewConfig;
-
-    /**
-     * The config variable data given in the constants. This is hold here, so the view
-     * config can be updated with this data.
-     */
-    data: any;
-}
+import { map } from 'rxjs/operators';
```
Comment (FinnStutzenstein) [inline client/src/app/core/repositories/config/config-repository.service.ts]: Order of imports

File: client/src/app/core/repositories/config/config-repository.service.ts [hunk 2]
```diff
@@ -160,151 +134,66 @@ export class ConfigRepositoryService extends BaseRepository<ViewConfig, Config,
...
+            this.configsSubject.next(this.configs);
         }
     }
 
-    /**
-     * Getter for the config structure
-     */
-    public getConfigStructure(): ConfigGroup[] {
-        return this.configs;
+    public getAvailableGroupsOberservable(): Observable<string[]> {
```
Comment (FinnStutzenstein) [inline client/src/app/core/repositories/config/config-repository.service.ts]: move to getters; docs

File: client/src/app/core/repositories/config/config-repository.service.ts [hunk 3]
```diff
@@ -160,151 +134,66 @@ export class ConfigRepositoryService extends BaseRepository<ViewConfig, Config,
...
             }
-        }
+        });
+
+        this.configsSubject.next(groups);
     }
 
     /**
-     * Saves a config value.
+     * Saves a config value. The server nneds the key instead of the id to fetch the config variable.
```
Comment (FinnStutzenstein) [inline client/src/app/core/repositories/config/config-repository.service.ts]: typo

File: client/src/app/site/config/components/config-list/config-list.component.ts [hunk 4]
```diff
@@ -1,10 +1,12 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { Title } from '@angular/platform-browser';
 
 import { TranslateService } from '@ngx-translate/core';
 
 import { ConfigRepositoryService, ConfigGroup } from 'app/core/repositories/config/config-repository.service';
 import { BaseComponent } from 'app/base.component';
+import { ActivatedRoute } from '@angular/router';
```
Comment (FinnStutzenstein) [inline client/src/app/site/config/components/config-list/config-list.component.ts]: order of imports

**Comment by FinnStutzenstein** [comment]
@GabrielInTheWorld Rebased and I added `/rest/core/config/bulk_update/` which accepts `[{key: <key>, value: <value>}, ...]` in a post request. It returns `{errors: {<key>: <error message>}}`, if one or more keys gave an error during update. Please make sure, that the assosicated fields show this message and the user will not be navigated away until resolving these errors.

File: openslides/core/views.py [hunk 5]
```diff
@@ -436,6 +451,40 @@ def update(self, request, *args, **kwargs):
...
+                raise ValidationError(
+                    {"detail": "Invalid input. Config value is missing."}
+                )
+
+        errors = {}
+        for entry in request.data:
+            try:
+                config[entry["key"]] = entry["value"]
+            except ConfigError as err:
+                errors[key] = str(err)
```
Comment (GabrielInTheWorld) [inline openslides/core/views.py]: The last error overwrites the previous errors. So only one error is returned, even if several fields contain errors.

Comment (FinnStutzenstein) [reply openslides/core/views.py]: Yes, it should be: errors[entry["key"]] = str(err)

**Comment by FinnStutzenstein** [comment]
Two small things, than merge.

**Comment by GabrielInTheWorld** [comment]
> * The tiles should be replaced with a simple (android like) list
The decision working with tiles is inspired by our designer.

> * instead of the "question mark-circle-icon" right next to the options, please use "mat-hint" (where possible) to display the description of the options
This was decided, because the hints are different number of lines long and make it unable to deal with it.

**Pull request merged**